src/core/mac/mac_header_ie.hpp

```diff
@@ -316,10 +313,118 @@ class ThreadIe
     static constexpr uint32_t kVendorOuiThreadCompanyId = 0xeab89b;
     static constexpr uint8_t  kEnhAckProbingIe          = 0x00;
 };
-#endif
 
-#endif // OPENTHREAD_CONFIG_TIME_SYNC_ENABLE || OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE ||
-       // OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE
+#if OPENTHREAD_CONFIG_WAKEUP_COORDINATOR_ENABLE || OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+/**
+ * This class implements Rendezvous Time IE data structure.
+ *
+ * IEEE 802.15.4 Rendezvous Time IE contains two fields, Rendezvous Time and
+ * Wake-up Interval, but the Wake-up Interval is not used in Thread, so it is
+ * not included in this class.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class RendezvousTimeIe
+{
+public:
+    static constexpr uint8_t kHeaderIeId    = 0x1d;
+    static constexpr uint8_t kIeContentSize = sizeof(uint16_t);
+
+    /**
+     * This method returns the Rendezvous Time.
+     *
+     * @returns the Rendezvous Time.
+     *
+     */
+    uint16_t GetRendezvousTime(void) const { return LittleEndian::HostSwap16(mRendezvousTime); }
+
+    /**
+     * This method sets the Rendezvous Time.
+     *
+     * @param[in]  aRendezvousTime  The Rendezvous Time.
+     *
+     */
+    void SetRendezvousTime(uint16_t aRendezvousTime) { mRendezvousTime = LittleEndian::HostSwap16(aRendezvousTime); }
+
+private:
+    uint16_t mRendezvousTime;
+} OT_TOOL_PACKED_END;
+
+/**
+ * Implements Connection IE data structure.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class ConnectionIe : public VendorIeHeader
+{
+public:
+    static constexpr uint8_t kHeaderIeId      = ThreadIe::kHeaderIeId;
+    static constexpr uint8_t kIeContentSize   = ThreadIe::kIeContentSize + sizeof(uint8_t);
+    static constexpr uint8_t kThreadIeSubtype = 0x01;
+
+    /**
+     * Initializes the Connection IE.
+     *
+     */
+    void Init(void)
+    {
+        SetVendorOui(ThreadIe::kVendorOuiThreadCompanyId);
+        SetSubType(kThreadIeSubtype);
+        mConnectionWindow = 0;
+    }
+
+    /**
+     * Returns the Retry Interval.
+     *
+     * The Retry Interval defines how frequently the Wake-up End Device is
+     * supposed to retry sending the Parent Request to the Wake-up Coordinator.
+     *
+     * @returns the Retry Interval in the units of Wake-up Intervals (7.5ms).
```

Comment:
The spec said that the default wake-up interval is 7.5 ms. Does it mean that the wake-up interval must be 7.5 ms? And the wake-up interval is configurable in your implementation (https://github.com/nrfconnect/sdk-openthread/commit/collab-thread-s2s#diff-dea0ae821fb87ed58bddcfd186b322363f2249d899f3e3d4a2d6a04613c1cd78R1170).

Reply:
Right, the spec allows overriding WAKEUP_INTERVAL and LISTEN_DURATION, but then WC and WED would have to have a way to agree upon these parameters out of band (since both values must correspond to each other to assure a successful wake-up), so I guess in most cases the defaults will be used. I can make it clearer in the comment though.